src/pdn/src/pdn.tcl

```diff
@@ -254,6 +266,54 @@ proc define_pdn_grid {args} {
   }
 }
 
+sta::define_cmd_args "define_power_switch_cell" {[-name <name>] \
+                                                 [-control <control_name>] \
+                                                 [-acknowledge <acknowledge_name>] \
+                                                 [-power_switchable <power_switchable_net_name>] \
+                                                 [-power <power_net_name> ] \
+                                                 [-ground <ground_net_name>] }
+
+proc define_power_switch_cell {args} {
+  pdngen::check_design_state
+
+  sta::parse_key_args "define_power_switch_cell" args \
+    keys {-name -control -acknowledge -power_switchable -power -ground} 
+
+  if {[llength $args] > 0} {
+    utl::error PDN 1182 "Unexpected argument [lindex $args 0] for define_power_switch_cell command."
+  }
+
+  if {![info exists keys(-name)]} {
+    utl::error PDN 1183 "The -name argument is required."
+  }
+
+  if {![info exists keys(-control)]} {
+    utl::error PDN 1184 "The -control argument is required."
+  }
+
+  if {![info exists keys(-acknowledge)]} {
+    utl::error PDN 1185 "The -acknowledge argument is required."
+  }
```

Comment:
Is this required? is it possible that there is no acknowledge pin?

Reply:
Agreed - acknowledge is optional